cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectConfigurationBuilder.java

```diff
@@ -0,0 +1,327 @@
+                    }
+
+                    if (oauth.getClientAssertion() != null) {
+                        jaasConfig.append(" oauth.client.assertion=${strimzienv:KAFKA_CONNECT_OAUTH_CLIENT_ASSERTION}");
+                    }
+
+                    if (oauth.getTlsTrustedCertificates() != null && !oauth.getTlsTrustedCertificates().isEmpty()) {
+                        jaasConfig.append(" oauth.ssl.truststore.location=\"/tmp/strimzi/oauth.truststore.p12\" oauth.ssl.truststore.password=" + PLACEHOLDER_CERT_STORE_PASSWORD_CONFIG_PROVIDER_ENV_VAR + " oauth.ssl.truststore.type=\"PKCS12\"");
+                    }
+
+                    jaasConfig.append(";");
+                    oauthCallbackClass = "sasl.login.callback.handler.class=io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler";
+                    producerOauthCallbackClass = "producer." + oauthCallbackClass;
+                    consumerOauthCallbackClass = "consumer." + oauthCallbackClass;
+                    adminOauthCallbackClass = "admin." + oauthCallbackClass;
+
+                }
+                writer.println("sasl.mechanism=" + saslMechanism);
+                writer.println("sasl.jaas.config=" + jaasConfig);
+                writer.println(oauthCallbackClass);
+
+                writer.println("producer.sasl.mechanism=" + saslMechanism);
+                writer.println("producer.sasl.jaas.config=" + jaasConfig);
+                writer.println(producerOauthCallbackClass);
+
+                writer.println("consumer.sasl.mechanism=" + saslMechanism);
+                writer.println("consumer.sasl.jaas.config=" + jaasConfig);
+                writer.println(consumerOauthCallbackClass);
+
+                writer.println("admin.sasl.mechanism=" + saslMechanism);
+                writer.println("admin.sasl.jaas.config=" + jaasConfig);
+                writer.println(adminOauthCallbackClass);
+
+                writer.println();
+            }
+        }
+        return this;
+    }
+
+    /**
+     * Adds consumer client rack {@code rack.id}. The rack ID will be set in the container based on the value of the
+     * {@code STRIMZI_RACK_ID} env var.
+     *
+     * @return Returns the builder instance
+     */
+    public KafkaConnectConfigurationBuilder withRackId()   {
+        printSectionHeader("Additional information");
+        writer.println("consumer.client.rack=${strimzienv:STRIMZI_RACK_ID}");
+        writer.println();
+        return this;
+    }
+
+    /**
+     * Configures the Kafka Connect configuration providers
+     *
+     * @param userConfig    the user configuration to extract the possible user-provided config provider configuration from it
+     */
+    public void configProviders(AbstractConfiguration userConfig) {
+        printSectionHeader("Config providers");
+        String strimziConfigProviders = "strimzienv,strimzifile";
```

Comment:
you are putting a "strimzifile" config provider here which is then not configured. I think this could be related to your question about having the file config provider here because used by Mirror Maker 2 anyway. I agree with you I would do so.

Reply:
Sorry I think I had left `strimzifile` here by mistake. I wanted to check with you first before configuring. I will then configure it properly here and remove it from MM2 default configurations.